src/module.c

```diff
@@ -6295,7 +6359,7 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
         }
         c = NULL; /* Make sure not to free the client */
     } else {
-        reply = moduleParseReply(c, ctx);
+        reply = moduleParseReply(c, (ctx->flags & REDISMODULE_CTX_AUTO_MEMORY) ? ctx : NULL);
```

Comment:
Seems like today we can not hold the call reply object for longer time then the context, in case there is no auto memory set I do not see a reason why not to allow it?

Reply:
so let's mention this in the top comment.

Reply:
Mentioned